### PR TITLE
kahip: update build system to cmake for v3.11, retain scons for older versions

### PR DESCRIPTION
**Pull request by ma595**

### CODE REVIEW

File: var/spack/repos/builtin/packages/kahip/package.py [hunk 1]
```diff
@@ -23,12 +23,14 @@ class Kahip(SConsPackage):
     """
 
     homepage  = 'http://algo2.iti.kit.edu/documents/kahip/index.html'
-    url       = 'http://algo2.iti.kit.edu/schulz/software_releases/KaHIP_2.00.tar.gz'
+    url       = 'https://github.com/KaHIP/KaHIP/archive/v3.11.tar.gz'
     git       = 'https://github.com/schulzchristian/KaHIP.git'
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/kahip/package.py]: Should this be updated to match the `github` repo above, as well?

**Comment by tgamblin** [comment]
@adamjstewart: this is a multi-build-system package -- just want to make sure it looks ok to you.

**Review by tgamblin — DISMISSED** [review]
No major requests -- just one note on the `git` URL.

File: var/spack/repos/builtin/packages/kahip/package.py (hunk 1 above)
Comment (ma595) [reply var/spack/repos/builtin/packages/kahip/package.py]: amended git url as suggested

File: var/spack/repos/builtin/packages/kahip/package.py [hunk 2]
```diff
@@ -41,9 +44,11 @@ class Kahip(SConsPackage):
     # *) first file patches Sconstruct files present in all versions (from
     # 2.00 to 2.11)
     # *) second is for files only present in 2.00
-    patch('fix-sconstruct-for-py3.patch', when='@2: ^python@3:')
+    patch('fix-sconstruct-for-py3.patch', when='@2:2.11 ^python@3:')
     patch('fix-sconstruct-for-py3-v2.00.patch', when='@2.00 ^python@3:')
 
+    # 'when' decorators to override new CMake build approach (old build was SConstruct).
+    @when("@:2.11")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/kahip/package.py]: Should these be 2.10 instead? The CMake build system was added in 2.11.

File: var/spack/repos/builtin/packages/kahip/package.py [hunk 3]
```diff
@@ -23,12 +23,15 @@ class Kahip(SConsPackage):
...
     homepage  = 'http://algo2.iti.kit.edu/documents/kahip/index.html'
-    url       = 'http://algo2.iti.kit.edu/schulz/software_releases/KaHIP_2.00.tar.gz'
-    git       = 'https://github.com/schulzchristian/KaHIP.git'
+    url       = 'https://github.com/KaHIP/KaHIP/archive/v3.11.tar.gz'
+    git       = 'https://github.com/KaHIP/KaHIP.git'
+    maintainers = ["ma595"]
 
     version('develop', branch='master')
+    version('3.11', sha256='347575d48c306b92ab6e47c13fa570e1af1e210255f470e6aa12c2509a8c13e3')
     version('2.00', sha256='1cc9e5b12fea559288d377e8b8b701af1b2b707de8e550d0bda18b36be29d21d')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/kahip/package.py]: This release isn't available on GitHub, can you add a `url='...'` to this version? Should we deprecate this version?

File: var/spack/repos/builtin/packages/kahip/package.py [hunk 4]
```diff
@@ -23,12 +23,15 @@ class Kahip(SConsPackage):
...
 
+    depends_on('scons @3.1.2 ', type='build', when='@2:2.11')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/kahip/package.py]: The package doesn't mention any specific required version of scons, I would remove `@3.1.2`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/kahip/package.py (hunk 2 above)
Comment (ma595) [reply var/spack/repos/builtin/packages/kahip/package.py]: You are correct. This change has been made.

File: var/spack/repos/builtin/packages/kahip/package.py (hunk 3 above)
Comment (ma595) [reply var/spack/repos/builtin/packages/kahip/package.py]: Do you mean v2.00? I've added a URL and deprecated=True

File: var/spack/repos/builtin/packages/kahip/package.py (hunk 4 above)
Comment (ma595) [reply var/spack/repos/builtin/packages/kahip/package.py]: Removed

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
It looks like there's a merge conflict (the kahip package must have been updated since you branched off of develop). Can you resolve the merge conflicts (either through the GitHub GUI or CLI)?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**